src/utils/logging_utils.py

```
@@ -0,0 +1,460 @@
+    """
+    Async context manager for automatic operation logging.
+    
+    Usage:
+        async with log_operation(ctx, "my_tool", "my_operation") as logger:
+            # Your operation code here
+            logger.info("Step completed")
+    """
+    logger = OperationMonitor().get_logger(tool_name)
+    operation_id = operation_name or f"{tool_name}_{int(time.time())}"
+    
+    metrics = logger.start_operation(operation_id, context_data)
```

Ruff (0.12.2)

Local variable `metrics` is assigned to but never used

Remove assignment to unused variable `metrics`

(F841)